### PR TITLE
add another destination type (DUMMY), that makes the whole propagatio…

### PR DESCRIPTION
**Pull request by mvocu**
…n sucessfull without actually running SEND task

* adds propagation type DUMMY to the Destination
* dispatcher now sets tasks with no destinations to the error state
* engine does not try to propagate tasks that have DUMMY destination, they are considered done
* engine sets tasks with no destinations to the error state

See RT#214867.

### CODE REVIEW

File: perun-core/src/main/java/cz/metacentrum/perun/core/impl/ServicesManagerImpl.java [hunk 1]
```diff
@@ -131,8 +131,11 @@ public Destination mapRow(ResultSet rs, int i) throws SQLException {
 			destination.setModifiedAt(rs.getString("destinations_modified_at"));
 			destination.setModifiedBy(rs.getString("destinations_modified_by"));
 			try { // do not mind if the column is not in the results
-				if(rs.getString("f_s_des_propagation_type").equals(Destination.PROPAGATIONTYPE_SERIAL)) {
-					destination.setPropagationType(Destination.PROPAGATIONTYPE_SERIAL);
+				String ptype = rs.getString("f_s_des_propagation_type");
+				if(ptype.equals(Destination.PROPAGATIONTYPE_SERIAL) ||
```
Comment (zlamalp) [inline perun-core/src/main/java/cz/metacentrum/perun/core/impl/ServicesManagerImpl.java]: In regards of need to display info in GUI, please add this optional reading of propagation type to ``RICH_DESTINATION_MAPPER`` too and update select statements to retrieve also ``f_s_des_propagation_type`` column.

Comment (mvocu) [reply perun-core/src/main/java/cz/metacentrum/perun/core/impl/ServicesManagerImpl.java]: Revisited, but please review this, especially in RPC.

File: perun-rpc/src/main/java/cz/metacentrum/perun/rpc/methods/ServicesManagerMethod.java [hunk 2]
```diff
@@ -757,16 +757,24 @@ public Destination call(ApiCaller ac, Deserializer parms) throws PerunException
 		public Destination call(ApiCaller ac, Deserializer parms) throws PerunException {
 			ac.stateChangingCheck();
 
+			Destination destination;
+			
+			if(parms.contains("propagationType")) {
```
Comment (zlamalp) [inline perun-rpc/src/main/java/cz/metacentrum/perun/rpc/methods/ServicesManagerMethod.java]: Implementation is correct, just please double javadoc -> for each existing javadoc add new with "propagationType" param present.

File: perun-rpc/src/main/java/cz/metacentrum/perun/rpc/methods/ServicesManagerMethod.java [hunk 3]
```diff
@@ -787,9 +795,17 @@ public Destination call(ApiCaller ac, Deserializer parms) throws PerunException
 		public List<Destination> call(ApiCaller ac, Deserializer parms) throws PerunException {
 			ac.stateChangingCheck();
 
+			Destination destination;
+
```
Comment (zlamalp) [inline perun-rpc/src/main/java/cz/metacentrum/perun/rpc/methods/ServicesManagerMethod.java]: Implementation is correct, just please double javadoc -> for each existing javadoc add new with "propagationType" param present.

**Comment by zlamalp** [comment]
Merged with local fixes.

**Pull request merged**